Feat/clean architecture

meczajkowski opened the pull request.

meczajkowski committed:
refactor: implement service orders data layer with DTOs and database abstraction

meczajkowski committed:
refactor: reorganize project structure by moving services to domains directory

meczajkowski committed:
refactor: restructure project architecture by moving domains to backend directory

mkokosinski commented on src/backend/service-orders/service-order.repository.ts:
```diff
@@ -0,0 +1,204 @@
+  findByIdWithRelations(
+    id: ServiceOrderWithRelationsDto['id'],
+  ): Promise<ServiceOrderWithRelationsDto>;
+  findAll(): Promise<Array<ServiceOrderDto>>;
+  findAllWithRelations(): Promise<Array<ServiceOrderWithRelationsDto>>;
```
```suggestion
  findAll(): Promise<ServiceOrderDto[]>;
  findAllWithRelations(): Promise<ServiceOrderWithRelationsDto[]>;
```

mkokosinski commented on src/backend/service-orders/service-order.repository.ts:
```diff
@@ -0,0 +1,204 @@
+  assignedTo: User | null;
+};
+
+const fromPrismaToDto = (serviceOrder: ServiceOrder): ServiceOrderDto => ({
+  id: serviceOrder.id,
+  troubleDescription: serviceOrder.troubleDescription,
+  status: serviceOrder.status as ServiceOrderStatus,
+  deviceId: serviceOrder.deviceId,
+  assignedToId: serviceOrder.assignedToId,
+  completedAt: serviceOrder.completedAt,
+  createdAt: serviceOrder.createdAt,
+  updatedAt: serviceOrder.updatedAt,
+});
```
```suggestion
const fromPrismaToDto = (serviceOrder: ServiceOrder): ServiceOrderDto => ({
 ...serviceOrder,
  status: serviceOrder.status as ServiceOrderStatus,
});
```

mkokosinski commented on src/backend/service-orders/service-order.repository.ts:
```diff
@@ -0,0 +1,204 @@
+const fromPrismaToDtoWithRelations = (
+  serviceOrder: ServiceOrderWithRelations,
+): ServiceOrderWithRelationsDto => {
+  return {
+    id: serviceOrder.id,
+    troubleDescription: serviceOrder.troubleDescription,
+    status: serviceOrder.status as ServiceOrderStatus,
+    device: {
+      id: serviceOrder.device.id,
+      serialNumber: serviceOrder.device.serialNumber,
+      model: serviceOrder.device.model,
+      customer: serviceOrder.device.customer
+        ? {
+            id: serviceOrder.device.customer.id,
+            name: serviceOrder.device.customer.name,
+            email: serviceOrder.device.customer.email,
+            phone: serviceOrder.device.customer.phone,
+            address: serviceOrder.device.customer.address,
+            createdAt: serviceOrder.device.customer.createdAt,
+            updatedAt: serviceOrder.device.customer.updatedAt,
+          }
+        : null,
+    },
+    assignedTo: serviceOrder.assignedTo
+      ? {
+          id: serviceOrder.assignedTo.id,
+          name: serviceOrder.assignedTo.name,
+          email: serviceOrder.assignedTo.email,
+          image: serviceOrder.assignedTo.image,
+          role: serviceOrder.assignedTo.role as UserRole,
+          createdAt: serviceOrder.assignedTo.createdAt,
+          updatedAt: serviceOrder.assignedTo.updatedAt,
+        }
+      : null,
+    completedAt: serviceOrder.completedAt,
+    createdAt: serviceOrder.createdAt,
+    updatedAt: serviceOrder.updatedAt,
+  };
+};
```
```suggestion
  return {
    ...serviceOrder,
    status: serviceOrder.status as ServiceOrderStatus,
    device: {
      ...serviceOrder.device,
      customer: serviceOrder.device.customer || null,
    },
    assignedTo: serviceOrder.assignedTo
      ? {
          ...serviceOrder.assignedTo,
          role: serviceOrder.assignedTo.role as UserRole,
        }
      : null,
  };
};
```

meczajkowski committed:
refactor: reorganize service-orders and devices domains for Clean Architecture

- Move service-orders and devices to domains folder for better domain separation
- Create common mappers for service-orders and devices entities
- Replace inline transformers with dedicated mapper functions
- Fix type definitions between ServiceOrderDto and DeviceWithRelationsDto
- Update imports across affected components

meczajkowski committed:
refactor: standardize user and device mappers with consistent transformation logic

- Update user mapper to remove null check and simplify type conversion
- Remove inline device and user transformation functions from repositories
- Import and use dedicated mapper functions from common mappers
- Ensure consistent mapping approach across different domain repositories

meczajkowski committed:
refactor: standardize contacts domain with clean architecture principles

- Update contacts service and repository to follow consistent domain architecture
- Implement DTOs and repository pattern for contacts
- Add authorization checks in service layer
- Remove direct Prisma usage from service methods
- Simplify actions by using new service and repository methods
- Remove deprecated schema and types files

meczajkowski committed:
refactor: consolidate repository patterns and remove redundant auth repository

- Remove deprecated auth repository
- Update auth service to use users repository
- Standardize repository methods across domains
- Remove redundant existence checks in repository update and delete methods
- Improve error handling and data fetching in auth service

meczajkowski committed:
refactor: enhance repository documentation with improved type safety and error handling

- Update repository interfaces with more precise documentation
- Add explicit error handling annotations to repository methods
- Standardize module documentation with consistent @module paths
- Improve type safety in repository return types
- Add explicit handling for empty result scenarios in repositories

meczajkowski committed:
refactor: enhance service domain documentation and error handling

meczajkowski committed:
feat: implement services initialization and dependency injection

- Create services.ts to centralize service and repository instantiation
- Inject dependencies for all domain services using constructor injection
- Ensure single point of service and repository creation
- Standardize service and repository initialization across domains

meczajkowski committed:
refactor: update contact and customer actions with new service imports and DTO patterns

- Migrate contact and customer actions to use centralized services import
- Update action methods to use new DTO structures
- Remove deprecated schema and types files
- Standardize action method signatures with consistent parameter handling
- Improve type safety in contact and customer form components
